### PR TITLE
Fix some more duplicate enum to_py utility code names

### PR DESCRIPTION
**Pull request by da-woods**
It's possible to declare an extern enum multiple times, in a way that makes the declaration name the same. Therefore, also prefix declaration names with the module qualified name where possible.

### CODE REVIEW

**Comment by da-woods** [comment]
I'm fairly sure this does fix https://github.com/cython/cython/issues/5860

**Comment by EdwardSro** [comment]
> I'm fairly sure this does fix #5860

Yes, it does fix the compilation. Thanks!
But now I see the following errors/warnings when I run our python tests (that uses the cython generated code):

`<frozen importlib._bootstrap>:241: UserWarning: enum class rdma_port_space not importable from pyverbs.librdmacm_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_send_flags not importable from pyverbs.libibverbs. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_wr_opcode not importable from pyverbs.libibverbs_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_send_flags not importable from pyverbs.libibverbs_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_ops_wr_opcode not importable from pyverbs.libibverbs_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_ops_flags not importable from pyverbs.libibverbs_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_wq_type not importable from pyverbs.libibverbs_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_values_mask not importable from pyverbs.libibverbs. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_flow_attr_type not importable from pyverbs.libibverbs. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_qp_type not importable from pyverbs.libibverbs_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_qp_state not importable from pyverbs.libibverbs_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class ibv_mtu not importable from pyverbs.libibverbs_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class mlx5dv_mkey_err_type not importable from pyverbs.providers.mlx5.libmlx5. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class mlx5dv_dr_matcher_layout_flags not importable from pyverbs.providers.mlx5.libmlx5. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.
<frozen importlib._bootstrap>:241: UserWarning: enum class mlx5dv_dc_type not importable from pyverbs.providers.mlx5.mlx5dv_enums. You are probably using a cpdef enum declared in a .pxd file that does not have a .py  or .pyx file.`

**Comment by da-woods** [comment]
> But now I see the following errors/warnings when I run our python tests (that uses the cython generated code):

That's definitely intentional. The change behind most of this is that when you do something like

```
def func():
    return SomeCpdefEnum.SomeValue
```

it now returns an instance of the enum class rather than just an `int`. However, for that to work it needs to be able to `import` the enum class. If it can't do that you'll get a warning and it'll just return an int.

What it's recommending you do is create a `libibverbs_enums.pyx` file (and compile it). It can be empty, but it'll allow the enum wrapper class to be importable.

**Comment by scoder** [comment]
> it'll allow the enum wrapper class to be importable.

That might also solve the original problem why the OP resorted to a mix of include+import.

**Comment by EdwardSro** [comment]
> 
> What it's recommending you do is create a `libibverbs_enums.pyx` file (and compile it). It can be empty, but it'll allow the enum wrapper class to be importable.

But I already have a file, named enums.pyx, that is a symbolic link to libibverbs_enums.pxd and from python code the user imports from enums.pyx and it works fine. Shouldn't this behavior be kept in Cython 3?

The same goes for other warnings like libibverbs.pxd. Eventually, the user doesn't directly import from "libibverbs.pyx" but for multiple .pyx files, which themselves cimports from libibverbs.pxd...

It's either I'm missing something or the new warnings are an unwanted consequence of an underlying behavioral change in Cython.

**Comment by EdwardSro** [comment]
@da-woods  @scoder is there a reason that the PR is not merged yet?

File: Cython/Compiler/PyrexTypes.py [hunk 1]
```diff
@@ -5444,19 +5444,26 @@ def type_list_identifier(types):
...
 _type_identifier_cache = {}
-def type_identifier_from_declaration(decl):
-    safe = _type_identifier_cache.get(decl)
+def type_identifier_from_declaration(decl, scope_name = ""):
+    key = (decl, scope_name)
+    safe = _type_identifier_cache.get(key)
     if safe is None:
         safe = decl
+        if scope_name:
+            safe = "%s_%s" % (scope_name, safe)
```
Comment (scoder) [inline Cython/Compiler/PyrexTypes.py:5461]: It feels like this should use scope mangling rather than simple concatenation.

**Comment by scoder** [comment]
> is there a reason that the PR is not merged yet?

Probably just the distraction by the unrelated discussion above. I left a comment.

**Comment by EdwardSro** [comment]
Will this fix be part of 3.0.9 or 3.1?

**Comment by da-woods** [comment]
3.0.x commit 018535038

**Pull request merged**